File: Data Structures & Algorithms/max-area-of-island/submission-1.cpp
class Solution {
public:
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int ans{0};
        for(size_t i{0}; i<grid.size();i++){
            for(size_t j{0}; j<grid[0].size();j++){
                ans = max(ans, getArea(grid,i,j));
            }
        }
        return ans;
    }

    int getArea(vector<vector<int>>&grid, size_t i, size_t j){
        if(i < 0 || j < 0 || i >= grid.size() || j >= grid[0].size())
            return 0;
        if(grid[i][j] == 0)
            return 0;
        grid[i][j] = 0;
        return 1 + getArea(grid,i,j-1) + getArea(grid,i,j+1) + getArea(grid,i - 1,j) + getArea(grid,i + 1,j);
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i,j = 0,0
        dc = defaultdict(int)
        maxLen = 0
        while j < len(s):    
            while dc[s[j]] > 0:
                dc[s[i]] -= 1
                i += 1
            maxLen = max(maxLen, j - i + 1)
            dc[s[j]] += 1
            j += 1
        
        return maxLen
            